src/server/ns_turn_maps.c

```diff
@@ -645,8 +652,14 @@ static addr_elem *addr_list_get(addr_list_header *slh, const ioa_addr *key) {
     for (size_t i = 0; i < slh->extra_sz; ++i) {
       addr_elem *elem = &(slh->extra_list[i]);
       if (elem->value) {
-        if (addr_eq(&(elem->key), key)) {
-          return elem;
+        if (port) {
+          if (addr_eq(&(elem->key), key)) {
+            return elem;
+          }
+        } else {
+          if (addr_eq_no_port(&(elem->key), key)) {
+            return elem;
+          }
         }
```

Comment:
Are you proposing to use a function pointer at the top of the function, and then call the function pointer?

This could work, let me know if we are thinking the same thing here.